internal/report/report.go

```diff
@@ -270,3 +272,17 @@ const (
 	ExitCodeLow      ExitCode = 101
 	ExitCodeInfo     ExitCode = 100
 )
+
+// collectSummaryMetrics collects the metrics corresponding to the summary.
+func collectSummaryMetrics(sum summary) {
+	metrics.Collect("excluded", sum.excluded)
+	if sum.count == nil {
+		metrics.Collect("vulnerabilities", nil)
+		return
+	}
+	countMetrics := make(map[string]int)
+	for key, value := range sum.count {
+		countMetrics[key.String()] = value
+	}
+	metrics.Collect("vulnerabilities", countMetrics)
```

Comment:
After doing some tests I think you can simplify this code to just: 

```suggestion
	metrics.Collect("vulnerabilities", sum.count)
```

If you implement `encoding.TextMarshaler` for `Severity` instead of `*Severity`. In other words,

https://github.com/adevinta/lava/blob/3d37036306f263784c372be9d83fa09ea7823916/internal/config/config.go#L223

Would become.

```go
func (s Severity) MarshalText() (text []byte, err error) { 
```

If this works fine, this function becomes so simple that I would remove it and inline the two calls to `metrics.Collect`.